glusterd2/commands/volumes/volume-create.go

```diff
@@ -63,7 +63,7 @@ func volumeCreateHandler(w http.ResponseWriter, r *http.Request) {
 
 	var req api.VolCreateReq
 	if err := restutils.UnmarshalRequest(r, &req); err != nil {
-		restutils.SendHTTPError(ctx, w, http.StatusUnprocessableEntity, err)
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, err)
```

Comment:
While you're fixing the issue, you can also improve error returned.

Pass `errors.ErrJSONParsingFailed` as error here as well and also at other places through-out the code where `restutils.UnmarshalRequest` is used.

Reply:
Other places to check:

```
[ppai@gd2-1 glusterd2]$ grep -nR "restutils.UnmarshalRequest" .
./glusterd2/commands/peers/addpeer.go:23:	if err := restutils.UnmarshalRequest(r, &req); err != nil {
./glusterd2/commands/peers/editpeer.go:23:	if err := restutils.UnmarshalRequest(r, &req); err != nil {
./glusterd2/commands/volumes/optiongroup-create.go:31:	if err := restutils.UnmarshalRequest(r, &req); err != nil {
./glusterd2/commands/volumes/volume-option.go:89:	if err := restutils.UnmarshalRequest(r, &req); err != nil {
./glusterd2/commands/volumes/volume-create.go:65:	if err := restutils.UnmarshalRequest(r, &req); err != nil {
./glusterd2/commands/volumes/volume-statedump.go:96:	if err := restutils.UnmarshalRequest(r, &req); err != nil {
./glusterd2/commands/volumes/volume-expand.go:45:	if err := restutils.UnmarshalRequest(r, &req); err != nil {
./glusterd2/commands/global/setglobaloptions.go:17:	if err := restutils.UnmarshalRequest(r, &req); err != nil {
./plugins/georeplication/rest.go:92:	if err := restutils.UnmarshalRequest(r, &req); err != nil {
./plugins/georeplication/rest.go:238:	if err := restutils.UnmarshalRequest(r, &req); err != nil {
./plugins/georeplication/rest.go:711:	if err := restutils.UnmarshalRequest(r, &req); err != nil {
./plugins/georeplication/rest.go:854:	if err := restutils.UnmarshalRequest(r, &req); err != nil {
./plugins/georeplication/rest.go:1084:	if err := restutils.UnmarshalRequest(r, &sshkeys); err != nil {
./plugins/device/rest.go:22:	if err := restutils.UnmarshalRequest(r, req); err != nil {
./plugins/events/rest.go:20:	if err := restutils.UnmarshalRequest(r, &req); err != nil {
./plugins/events/rest.go:59:	if err := restutils.UnmarshalRequest(r, &req); err != nil {
```